python/data_structures/linked_list.py

```diff
@@ -38,6 +38,73 @@ def includes(self, value):
             current = current.next
         return False
 
+    def append(self, new_value):
+        """
+        Append stores passed in element value as new_value, and calls new_node method with new_value as arg
+        If self.head == None we are at the end of the linked list, and can save our new node here.
+        Else we save the value of self.head as held_value, so we dont lose the element stored there.
+        while we have values in the nodes, or .next does not = None, reassign held_value to held_value.next
+        and traverse to the next node, until we have self.node == None.
+        """
+        new_node = Node(new_value)
+        if self.head is None:
+            self.head = new_node
+            return
+        else:
+            held_value = self.head
+            while held_value.next is not None:
+                held_value = held_value.next
+            held_value.next = new_node
+
+    def insert_before(self, value, new_value):
+        """
+        make and store our new node
+        iterate through our list, looking for our before. Storing the element previous to it each time.
+        once we found the elem which equals our before, we can iterate through the list until we find its
+        previous elem, from there we can reassign next to our new node, and insert our new node, with a .next
+        equal to the before elem.
+
+        """
+        new_node = Node(new_value)  # new node
+        current = self.head
+
+        if self.head is None:
+            raise TargetError
+
+        if self.includes(value) is False:
+            raise TargetError
+
+        if current.value is value:  # if we are at the node we want to insert before
+            new_node.next = self.head  # new_node.next = the current node
+            self.head = new_node  # set our new node as head
+
+        while current.next is not None:
+            if current.next.value is value:  # if the next node is what we want to insert before
+                new_node.next = current.next  # reassign next pointer of new node, to the item to insert before
+                current.next = new_node  # reassign our current node as our newly formatted new node
+                return
+            else:
+                current = current.next  # traverse to the next node
+
+    def insert_after(self, value, new_value):
+        """
+        inserts a node after a given value
+        """
+
+        if self.head is None:
+            raise TargetError
+
+        if self.includes(value) is False:
+            raise TargetError
+
+        new_node = Node(new_value)
+        current = self.head
+        while current:
+            if current.value is value:
+                new_node.next = current.next
+                current.next = new_node
+                break
```

Comment:
Your insert_after method doesn't actually traverse the linked list, and looks like it will be caught in an infinite loop if `value` is not the value of the head node.